Source/CesiumRuntime/Private/CesiumTextureUtility.cpp

```diff
@@ -670,4 +762,93 @@ TextureAddress convertGltfWrapTToUnreal(int32_t wrapT) {
   }
 }
 
+std::optional<CesiumAsync::SharedFuture<CesiumGltf::ImageCesium*>>
+createMipMapsForSampler(
+    const CesiumAsync::AsyncSystem& asyncSystem,
+    const CesiumGltf::Sampler& sampler,
+    CesiumGltf::ImageCesium& image) {
+  std::unique_lock lock(textureResourceMutex);
+
+  ExtensionUnrealTextureResource& extension =
+      image.addExtension<ExtensionUnrealTextureResource>();
+
+  // Future already exists, we don't need to do anything else.
+  if (extension.preprocessFuture.has_value()) {
+    return extension.preprocessFuture.value();
+  }
+
+  // Generate mipmaps if needed.
+  // An image needs mipmaps generated for it if:
+  // 1. It is used by a Texture that has a Sampler with a mipmap filtering
+  // mode, and
+  // 2. It does not already have mipmaps.
+  // It's ok if an image has mipmaps even if not all textures will use them.
+  // There's no reason to have two RHI textures, one with and one without
+  // mips.
+  bool needsMipmaps;
```

Comment:
Could replace with a call to `getUseMipmapsIfAvailableFromSampler` now that you've factored that out.